src/query/service/src/pipelines/executor/pipeline_executor.rs

```diff
@@ -397,14 +399,17 @@ impl Drop for PipelineExecutor {
     fn drop(&mut self) {
         self.finish(None);
 
-        let mut guard = self.on_finished_callback.lock();
-        if let Some(on_finished_callback) = guard.take() {
-            drop(guard);
-            let cause = Some(ErrorCode::Internal(
-                "Pipeline illegal state: not successfully shutdown.",
-            ));
-            if let Err(cause) = catch_unwind(move || on_finished_callback(&cause)).flatten() {
-                warn!("Pipeline executor shutdown failure, {:?}", cause);
+        let inited = *self.inited.lock();
+        if inited {
```

Comment:
This will cause to a session leak. we must call 'on_finished'.

Reply:
why session leak? drop session in a callback?

Reply:
is it ok to add on_finished to  PipelineCompleteExecutor/PipelinePullingExecutor, and call it here

```
impl PipelineCompleteExecutor {
   ...
    pub fn on_finished(&self, cause: Option<ErrorCode>) {
        self.executor.on_finished(&cause).ok();
    }
}

```